### PR TITLE
Added an example and doc of component overrides in markdown

### PR DESCRIPTION
**Pull request by oorestisime**
Closes: #2574

Signed-off-by: Orestis Ioannou <oorestisime@gmail.com>

#### What does this PR do?
Adds an example and documentation on markdown component overrides

#### How should this be manually tested?
stoirybook
#### What are the relevant issues?
#2574
#### Is this change backwards compatible or is it a breaking change?
no code changes here. only doc and storybook

### CODE REVIEW

File: src/js/components/Markdown/doc.js [hunk 1]
```diff
@@ -11,9 +11,12 @@ export const doc = Markdown => {
       <Markdown>{content}</Markdown>`,
     );
 
-  // DocumentedMarkdown.propTypes = {
-  //   content: PropTypes.string.description('The markdown text to render.'),
```
Comment (ShimiSun) [inline src/js/components/Markdown/doc.js]: I don't see the content anywhere, have we decided it is no longer needed?

File: src/js/components/Markdown/doc.js [hunk 2]
```diff
@@ -11,9 +11,12 @@ export const doc = Markdown => {
...
-  // };
+  DocumentedMarkdown.propTypes = {
+    components: PropTypes.objectOf(PropTypes.element).description(
+      `Custom components to override default html tags such as 'img' or 'pre'.
+By default only 'p' and 'a' are overrided with the Paragraph and Anchor components`,
```
Comment (ShimiSun) [inline src/js/components/Markdown/doc.js]: I think it should be `override`

File: src/js/components/Markdown/markdown.stories.js [hunk 3]
```diff
@@ -21,4 +26,18 @@ const SimpleMarkdown = () => (
   </Grommet>
 );
 
-storiesOf('Markdown', module).add('Simple Markdown', () => <SimpleMarkdown />);
+const StyledPre = styled.pre`
+  background-color: #7d4cdb;
+`;
+
+const ComponentOverrideMarkdown = () => (
```
Comment (ShimiSun) [inline src/js/components/Markdown/markdown.stories.js]: Thank you for adding a story!

File: src/js/components/Markdown/doc.js (hunk 1 above)
Comment (oorestisime) [reply src/js/components/Markdown/doc.js]: Well it was commented out so i thought it should be deleted. If you prefer i can let the comment there.

File: src/js/components/Markdown/doc.js (hunk 2 above)
Comment (oorestisime) [reply src/js/components/Markdown/doc.js]: or maybe `are overriden` ?

**Pull request merged**